lib/internal/webstreams/readablestream.js

```diff
@@ -1869,6 +1870,7 @@ function readableStreamCancel(stream, reason) {
 function readableStreamClose(stream) {
   assert(stream[kState].state === 'readable');
   stream[kState].state = 'closed';
+  stream[kState].streamClosed?.resolve?.();
```

Comment:
Can streamClosed be nully?

Reply:
I think it can because initially when I tested without null checks it failed

Reply:
Then I think there might be a bug somewhere. What does it mean when it's nully?

Reply:
Okay trying to investigate

Reply:
Fixed this, removed the null checks basically it was breaking tests where tee() was being used to construct streams